internal/mode/static/usage/secret.go

```diff
@@ -0,0 +1,60 @@
+package usage
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . SecretStorer
+
+// SecretStorer should store the usage Secret that contains the credentials for NGINX Plus usage reporting.
+type SecretStorer interface {
```

Comment:
I wonder if this should be more generic and be in the framework. 
We can use a generic here so that it can be a storer for any type that is client.Object.
However, we will need to move out credentials parsing logic

@kate-osborn @sjberman

Reply:
Do we currently have another `client.Object` that we store like this?

Reply:
I'm not aware of another store like this. The only reason this exists is because the Secret needs to be shared between the handler and the reporter.

There is another way to do this where the handler updates the reporter directly, but I didn't want the handler to be aware of the reporter. Figured having a shared object makes more sense.

Reply:
I think it makes sense too. @pleshakov since there's no other use case for this type, I think we should punt on making it more generic.

Reply:
>  I think we should punt on making it more generic.

sounds good